src/Line.php

```diff
@@ -428,4 +458,27 @@ public function insertJsonKey($key)
 
         return array_key_exists($key, $insert) ? $insert[$key] : false;
     }
+
+    /**
+     * Get the line's input in a safe way.
+     * 
+     * Escaping for html is done if this wasn't done by a previous listener already.
+     * Escaping is skipped is not enabled.
+     * 
+     * @since 1.2.0
+     * 
+     * @return string
+     */
+    public function escapedInput()
+    {
+        if ($this->lexer->escapeInput === false) {
+            return $this->input;
+        }
+        
+        if ($this->isEscaped()) {
+            return $this->input;
+        }
+        
+        return Listener::escape($this->input);
```

Comment:
As we have access to the lexer object, we could also create a public method in lexer to run escape, this would self determine whether encoding is enabled or not?

```php
public function escape($value)
{
   if (!$this->escapeInpute) {
      return $value;
   }

   return htmlspecialchars()
}
```

this would also allow a proper configuration of the html special chars encoding via lexer object, as static variables are configurable only in very ugly way => https://3v4l.org/HFYAk we could just us public variables in lexer.

So therefore developers could rely on escapedInput() method within process() without checking whether its enable or not (like you did), as the lexer object does this already.

Reply:
Good one. Fixed